docs/reference/stage-deploys-continuous-integration.md

```diff
@@ -0,0 +1,53 @@
+
+Deploying artifacts to stage follows the following steps:
+1. Once a new pull-request gets created in bigquery-etl, CI will [pull in the `generated-sql` branch](https://github.com/mozilla/bigquery-etl/blob/06d7baa3678509abc42ab190c6f1beabc920001c/.circleci/config.yml#L367-L372) to [determine all files that show any changes compared to what is deployed in production](https://github.com/mozilla/bigquery-etl/blob/06d7baa3678509abc42ab190c6f1beabc920001c/.circleci/config.yml#L373-L384) (it is assumed that the `generated-sql` branch reflects the artifacts currently deployed in production). All of these changed artifacts (UDFs, tables and views) will be deployed to the stage environment.
+    * This CI step runs after the `generate-sql` CI step to ensure that checks will also be executed on generated queries and to ensure `schema.yaml` files have been automatically created for queries. 
+2. The `bqetl` CLI has a command to run stage deploys, which is called in the CI: `./bqetl stage deploy --dataset-suffix=$CIRCLE_SHA1 $FILE_PATHS`
+    * `--dataset-suffix` will result in the artifacts being deployed to datasets that are suffixed by the current commit hash. This is to prevent any conflicts when deploying changes for the same artifacts in parallel and helps with debugging deployed artifacts.
+3. For every artifacts that gets deployed to stage all dependencies need to be determined and deployed to the stage environment as well since the stage environment doesn't have access to production. Before these artifacts get actually deployed, they need to be determined first by traversing artifact definitions.
+    * Determining dependencies is only relevant for UDFs and views. For queries, available `schema.yaml` files will simply be deployed. 
+    * For UDFs, if a UDF does call another UDF then this UDF needs to be deployed to stage as well.
+    * For views, if a view references another view, table or UDF then each of these referenced artifacts needs to be available on stage as well, otherwise the view cannot even be deployed to stage.
+    * If artifacts are referenced that are not defined as part of the bigquery-etl repo (like stable or live tables) then their schema will get determined and a placeholder `query.sql` file will be created
+    * Also dependencies of dependencies need to be deployed, and so on
+4. Once all artifacts that need to be deployed have been determined, all references to these artifacts in existing SQL files need to be updated. These references will need to point to the stage project and the temporary datasets that artifacts will be published to.
```

Comment:
Can you clarify how the list of artifacts is determined?

Reply:
I added some more context